src/main/java/org/elasticsearch/index/translog/TranslogFile.java

```diff
@@ -29,18 +30,28 @@
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
-public abstract class TranslogFile extends ChannelReader {
+public class TranslogFile extends ChannelReader {
 
     protected final ShardId shardId;
     protected final ReleasableLock readLock;
     protected final ReleasableLock writeLock;
+    protected volatile long lastSyncPosition = 0;
+    protected volatile int operationCounter;
+    protected volatile long lastPosition;
+    protected volatile long lastWrittenPosition;
 
```

Comment:
Are all these positions defined here versus being methods (e.g. getFilePointer) just because the class was changed to be no longer abstract? Why do we need to track 3 of them ourselves and what are there differences (especially lastPosition since its name is vague). Can we add comments?